File: docs/07-archive/legacy-repos/ebs_bo/src/bo/db/models/event_flight.py
from sqlmodel import SQLModel, Field

from .base import utcnow


class EventFlight(SQLModel, table=True):
    __tablename__ = "event_flights"
    event_flight_id: int | None = Field(default=None, primary_key=True)
    event_id: int = Field(foreign_key="events.event_id")
    display_name: str = Field(nullable=False)
    start_time: str | None = None
    is_tbd: bool = Field(default=False)
    entries: int = Field(default=0)
    players_left: int = Field(default=0)
    table_count: int = Field(default=0)
    status: str = Field(default="created")
    play_level: int = Field(default=1)
    remain_time: int | None = None
    source: str = Field(default="manual")
    synced_at: str | None = None
    created_at: str = Field(default_factory=utcnow)
    updated_at: str = Field(default_factory=utcnow)

File: team1-frontend/lib/models/entities/event_flight.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'event_flight.freezed.dart';
part 'event_flight.g.dart';

@freezed
class EventFlight with _$EventFlight {
  const factory EventFlight({
    @JsonKey(name: 'eventFlightId') required int eventFlightId,
    @JsonKey(name: 'eventId') required int eventId,
    @JsonKey(name: 'displayName') @Default('') String displayName,
    @JsonKey(name: 'startTime') String? startTime,
    @JsonKey(name: 'isTbd') @Default(false) bool isTbd,
    @Default(0) int entries,
    @JsonKey(name: 'playersLeft') @Default(0) int playersLeft,
    @JsonKey(name: 'tableCount') @Default(0) int tableCount,
    @Default('created') String status,
    @JsonKey(name: 'playLevel') @Default(1) int playLevel,
    @JsonKey(name: 'remainTime') int? remainTime,
    @Default('api') String source,
    @JsonKey(name: 'syncedAt') String? syncedAt,
    @JsonKey(name: 'createdAt') String? createdAt,
    @JsonKey(name: 'updatedAt') String? updatedAt,
    @JsonKey(name: 'playerCount') int? playerCount,
  }) = _EventFlight;

  factory EventFlight.fromJson(Map<String, dynamic> json) =>
      _$EventFlightFromJson(json);
}
